File: js/tlg/state/StatePaused.ts
import { State } from './State';
import { InternalContext } from '../context/InternalContext';
import { PlatformContext } from '../context/PlatformContext';
import { StateID } from './StateID';
import { StorageContext } from '../context/StorageContext';

export class StatePaused extends State {
    
    constructor(c:InternalContext, s:StorageContext) {
        super(c, s);
        this.context.setStatusText("Paused");
    }


    public getID(): number {
        return StateID.ID_PAUSED;
    }

    
    public togglePause(c:PlatformContext):StateID {
        return new StateID(StateID.ID_RUNNING);
    }
}

File: tlg/src/main/kotlin/tlg/state/StatePaused.kt
package tlg.state

import tlg.context.InternalContext
import tlg.context.PlatformContext


class StatePaused(pContext: InternalContext) : State(pContext) {
    override fun init(pContext: PlatformContext): State {
        return this
    }

    override fun togglePause(pContext: PlatformContext): State {
        return StateRunning(iContext)
    }

    override val id = ID

    override fun toString(): String {
        return "Paused"
    }

    companion object {
        const val ID = 2
    }
}
